helper/resource/testing_config.go

```diff
@@ -101,10 +101,12 @@ func testStep(
 	}
 
 	// And another after a Refresh.
-	state, err = ctx.Refresh()
-	if err != nil {
-		return state, fmt.Errorf(
-			"Error on follow-up refresh: %s", err)
+	if !step.Destroy || (step.Destroy && !step.PreventPostDestroyRefresh) {
```

Comment:
Oh, sorry... I re-read your description and I think I understand what you were going for here now: the refresh fails because the bucket (managed by Terraform in this case) is deleted.

I wonder if this would be necessary if your `aws_s3_object` data block interpolated the bucket name and object key from the two resources... I expect that in that case Terraform sould understand that the data source depends on the bucket and the object and would thus *not* try to refresh them when the resources are not present. Unfortunately I expect that this will not work as desired, because the bucket name and object key are not computed in the configuration so Terraform will think it "knows" these values even before those objects are created. This seems like a bug but I'm not sure what the *right* behavior would be given the building blocks this is made of.